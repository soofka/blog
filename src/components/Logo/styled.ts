import styledComponents from 'styled-components';

export const StyledLogo = styledComponents.img`
  display: block;
  margin: ${props => props.theme.marginL} auto;
  padding: 0;
  width: ${props => 28 * props.theme.baseDistance}px;
  height: ${props => 28 * props.theme.baseDistance}px;
  border-radius: ${props => 4 * props.theme.baseDistance}px;

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    float: left;
    margin: 0;
    padding: 0;
    width: ${props => 8 * props.theme.baseDistance}px;
    height: ${props => 8 * props.theme.baseDistance}px;
    border-radius: ${props => props.theme.baseDistance}px;
  }
`;

export default StyledLogo;
