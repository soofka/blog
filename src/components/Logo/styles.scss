@import 'components/Blog/variables.scss';

$logoMobileSize: 8 * $baseDistance;
$logoDesktopSize: 28 * $baseDistance;

.logo {
  display: block;
  margin: $marginL auto;
  padding: 0;
  width: $logoDesktopSize;
  height: $logoDesktopSize;
  border-radius: $logoDesktopSize / 8;
}

@media (max-width: $mobileBreakPointWidth) {
  .logo {
    float: left;
    margin: 0;
    padding: 0;
    width: $logoMobileSize;
    height: $logoMobileSize;
    border-radius: $logoMobileSize / 8;
  }
}
