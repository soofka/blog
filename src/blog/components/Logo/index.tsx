import * as React from 'react';
import { Link } from 'react-router-dom';

import { Label } from 'blog/components/Label';
import { getHomeBaseUrl, getImageFilePath } from 'common/helpers';

import { StyledLogoHeader, StyledLogoImage, StyledLogoImageContainer } from './styled';

const LOGO_ITEMS_COUNT = 4;

interface LogoPropsInterface {
  number?: number;
}

export const Logo = (props: LogoPropsInterface): JSX.Element => {
  const { number = Math.ceil(Math.random() * LOGO_ITEMS_COUNT) } = props;
  return (
    <div>
      <StyledLogoImageContainer>
        <Link to={getHomeBaseUrl()}>
          <StyledLogoImage src={getImageFilePath(`soofka-logo-${number}.jpg`)} />
        </Link>
      </StyledLogoImageContainer>
      <Link to={getHomeBaseUrl()}>
        <StyledLogoHeader>
          <Label name="page_title" />
        </StyledLogoHeader>
      </Link>
    </div>
  );
};
