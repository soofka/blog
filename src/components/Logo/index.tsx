import * as React from 'react';
import { Link } from 'react-router-dom';

import AssetsProvider from 'common/AssetsProvider';
import RoutingProvider from 'common/RoutingProvider';

import './styles.scss';

export const Logo = (): JSX.Element => {
  return (
    <Link to={RoutingProvider.getHomeNiceUrlBasePath()}>
      <img src={`${AssetsProvider.getImagesBasePath()}/soofka-logo.jpg`} className="logo" />
    </Link>
  );
};

export default Logo;
