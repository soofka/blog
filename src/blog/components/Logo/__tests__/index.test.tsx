import * as React from 'react';
import { shallow } from 'enzyme';

import Logo from 'components/Logo';

describe('<Logo/>', () => {

  it('renders correctly', () => {
    expect(shallow(
      <Logo
        number="1"
      />,
    )).toMatchSnapshot();
  });

});
