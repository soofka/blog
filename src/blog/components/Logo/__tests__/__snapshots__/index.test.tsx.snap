// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> renders correctly 1`] = `
<Link
  replace={false}
  to="/"
>
  <styled.img
    className="logo"
    src="/images/soofka-logo-1.jpg"
  />
</Link>
`;
